front/pages/api/w/[wId]/invitations/index.ts

```diff
@@ -152,7 +158,48 @@ async function handler(
         });
       }
       const existingMembers = await getMembers(auth);
-
+      const unconsumedInvitations = await MembershipInvitation.findAll({
+        where: {
+          workspaceId: owner.id,
+          status: ["pending", "revoked"],
+          createdAt: {
+            [Op.gte]: new Date(new Date().setHours(0, 0, 0, 0)),
+          },
+        },
+      });
+      if (unconsumedInvitations.length > MAX_UNCONSUMED_INVITATIONS) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Too many unconsumed invitations. Please ask your members to consume their invitations before sending more.`,
+          },
+        });
+      }
+      const emailsWithRecentUnconsumedInvitations = new Set(
+        unconsumedInvitations
+          .filter(
+            (i) =>
+              i.createdAt.getTime() >
+              new Date().getTime() - UNCONSUMED_INVITATION_COOLDOWN_PER_EMAIL_MS
+          )
+          .map((i) => i.inviteEmail.toLowerCase().trim())
+      );
+      if (
```

Comment:
Assuming one email address already has a fresh recent invitation, the entire operation will fail?

Reply:
yes, would you prefer that we only filter those out ?

Reply:
I mean as long as we have an explicit error message in the UI and the user does not lose its email addresses list, I'm fine.